### PR TITLE
Removed unnecessary validation

### PR DESCRIPTION
**Pull request by krishnakanth12**
Master API- Modified to allow some special characters in-class group, class name, class sections

### CODE REVIEW

File: eCampusAPI/ecampus_api/master/serializers.py [hunk 1]
```diff
@@ -1,280 +1,274 @@
...
+            raise serializers.ValidationError("Invalid institution name")
+        if Profile.objects.filter(trust_name=validated_data['trust_name']).exists():
+            raise serializers.ValidationError('trust name already exist.')
+        if Profile.objects.filter(mobile_number=validated_data['mobile_number']).exists():
+            raise serializers.ValidationError('mobile number already exist.')
+        if Profile.objects.filter(institution_name=validated_data['institution_name']).exists():
+            raise serializers.ValidationError('institution name already exist.')
+        if Profile.objects.filter(phone_number=validated_data['phone_number']).exists():
+                raise serializers.ValidationError('phone number already exist.')
+        return validated_data
```
Comment (kishoreecampus) [inline eCampusAPI/ecampus_api/master/serializers.py:34]: Remove this and use Unique attribute in model level.

File: eCampusAPI/ecampus_api/master/serializers.py [hunk 2]
```diff
@@ -1,280 +1,274 @@
...
+            raise serializers.ValidationError("Invalid institution name")
+        if Profile.objects.filter(trust_name=validated_data['trust_name']).exists():
+            raise serializers.ValidationError('trust name already exist.')
+        if Profile.objects.filter(mobile_number=validated_data['mobile_number']).exists():
+            raise serializers.ValidationError('mobile number already exist.')
+        if Profile.objects.filter(institution_name=validated_data['institution_name']).exists():
+            raise serializers.ValidationError(
+                'institution name already exist.')
+        if Profile.objects.filter(phone_number=validated_data['phone_number']).exists():
+            raise serializers.ValidationError('phone number already exist.')
```
Comment (kishoreecampus) [inline eCampusAPI/ecampus_api/master/serializers.py:58]: Remove this and use model validation

File: eCampusAPI/ecampus_api/master/serializers.py [hunk 3]
```diff
@@ -1,280 +1,274 @@
...
+        fields = ['class_group']
+    
+    def validate(self, validated_data):
+        if ClassGroup.objects.filter(class_group=validated_data['class_group']).exists():
+            raise serializers.ValidationError('class group already exist.')
+        if len(validated_data['class_group']) <= 2:
+            raise serializers.ValidationError("Invalid class group name")
+    
+        if validated_data['class_group'] and not validated_data['class_group'].isascii():
+            raise serializers.ValidationError("Invalid class group name")
```
Comment (kishoreecampus) [inline eCampusAPI/ecampus_api/master/serializers.py:94]: Remove this and use model validation

File: eCampusAPI/ecampus_api/master/serializers.py [hunk 4]
```diff
@@ -1,280 +1,274 @@
...
+        model = ClassGroup
+        fields = ['class_group','created_on'] 
+
+    def validate(self, validated_data):
+        if validated_data['class_group'] and not validated_data['class_group'].isascii():
+            raise serializers.ValidationError("Invalid class group name")
+        if len(validated_data['class_group']) <= 2:
+            raise serializers.ValidationError("Invalid class group name")
+        if ClassGroup.objects.filter(class_group=validated_data['class_group']).exists():
+            raise serializers.ValidationError('class group already exist.')
```
Comment (kishoreecampus) [inline eCampusAPI/ecampus_api/master/serializers.py:111]: Remove this and use model validation

File: eCampusAPI/ecampus_api/master/serializers.py [hunk 5]
```diff
@@ -1,280 +1,274 @@
...
+
+
+class ClassNameCreateSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = ClassName
+        fields = ['class_group','class_name']
+
+    def validate(self, validated_data):
+        if ClassName.objects.filter(class_name=validated_data['class_name']).exists():
+            raise serializers.ValidationError('class name already exist.')
```
Comment (kishoreecampus) [inline eCampusAPI/ecampus_api/master/serializers.py:137]: Remove this and use model validation

File: eCampusAPI/ecampus_api/master/serializers.py [hunk 6]
```diff
@@ -1,280 +1,274 @@
...
+        if validated_data['class_name'] and not validated_data['class_name'].isalnum():
+            raise serializers.ValidationError("Invalid class name")
```
Comment (kishoreecampus) [inline eCampusAPI/ecampus_api/master/serializers.py:139]: use isalpha function

File: eCampusAPI/ecampus_api/master/serializers.py [hunk 7]
```diff
@@ -1,280 +1,274 @@
...
+class ClassNameUpdateSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = ClassName
+        fields = ['class_group','class_name','created_on']
+    
+    def validate(self, validated_data):
+        if ClassName.objects.filter(class_name=validated_data['class_name']).exists():
+            raise serializers.ValidationError('class name already exist.')
+        if validated_data['class_name'] and not validated_data['class_name'].isalnum():
+            raise serializers.ValidationError("Invalid class name")
```
Comment (kishoreecampus) [inline eCampusAPI/ecampus_api/master/serializers.py:152]: Check on this.

File: eCampusAPI/ecampus_api/master/serializers.py [hunk 8]
```diff
@@ -1,280 +1,274 @@
...
+
+
+class SectionCreateSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Section
+        fields = ['class_name','section_name']
+
+    def validate(self,validated_data):
+        if validated_data['section_name'] and not validated_data['section_name'].isalpha():
+            raise serializers.ValidationError("Invalid section name")
```
Comment (kishoreecampus) [inline eCampusAPI/ecampus_api/master/serializers.py:176]: Check on this

File: eCampusAPI/ecampus_api/master/serializers.py [hunk 9]
```diff
@@ -1,280 +1,274 @@
...
+
+
+class CategoryCreateSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Category
+        fields = ['category']
+
+    def validate(self,validated_data):
+        if Category.objects.filter(category=validated_data['category']).exists():
+            raise serializers.ValidationError('Category already exists')
```
Comment (kishoreecampus) [inline eCampusAPI/ecampus_api/master/serializers.py:211]: Use unique attribute

File: eCampusAPI/ecampus_api/master/serializers.py [hunk 10]
```diff
@@ -1,280 +1,274 @@
...
+
+
+class CategoryUpdateSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Category
+        fields = ['category', 'created_on']
+
+    def validate(self, validated_data):
+        if Category.objects.filter(category=validated_data['category']).exists():
+            raise serializers.ValidationError('Category already exists')
```
Comment (kishoreecampus) [inline eCampusAPI/ecampus_api/master/serializers.py:225]: Use unique

**Review by kishoreecampus — CHANGES_REQUESTED** [review]

File: eCampusAPI/ecampus_api/master/serializers.py (hunk 1 above)
Comment (krishnakanth12) [reply eCampusAPI/ecampus_api/master/serializers.py]: removed please look out in models

File: eCampusAPI/ecampus_api/master/serializers.py (hunk 2 above)
Comment (krishnakanth12) [reply eCampusAPI/ecampus_api/master/serializers.py]: removed

File: eCampusAPI/ecampus_api/master/serializers.py (hunk 3 above)
Comment (krishnakanth12) [reply eCampusAPI/ecampus_api/master/serializers.py]: removed

File: eCampusAPI/ecampus_api/master/serializers.py (hunk 4 above)
Comment (krishnakanth12) [reply eCampusAPI/ecampus_api/master/serializers.py]: removed

File: eCampusAPI/ecampus_api/master/serializers.py (hunk 5 above)
Comment (krishnakanth12) [reply eCampusAPI/ecampus_api/master/serializers.py]: removed

File: eCampusAPI/ecampus_api/master/serializers.py (hunk 6 above)
Comment (krishnakanth12) [reply eCampusAPI/ecampus_api/master/serializers.py]: as discussed created a custom regex validator

File: eCampusAPI/ecampus_api/master/serializers.py (hunk 7 above)
Comment (krishnakanth12) [reply eCampusAPI/ecampus_api/master/serializers.py]: done

File: eCampusAPI/ecampus_api/master/serializers.py (hunk 8 above)
Comment (krishnakanth12) [reply eCampusAPI/ecampus_api/master/serializers.py]: removed

File: eCampusAPI/ecampus_api/master/serializers.py (hunk 9 above)
Comment (krishnakanth12) [reply eCampusAPI/ecampus_api/master/serializers.py]: yeah created in models

File: eCampusAPI/ecampus_api/master/serializers.py (hunk 10 above)
Comment (krishnakanth12) [reply eCampusAPI/ecampus_api/master/serializers.py]: yeah created please look out models

**Review by kishoreecampus — APPROVED** [review]

**Pull request merged**